src/popup/components/Register/index.tsx

```diff
@@ -66,53 +69,34 @@ const ActionContainer = styled.div`
 type RegisterProps = {
   loading: boolean;
   onNext: (password: string) => void;
-  onClickTerms: () => void;
-  onClickPrivacyPolicy: () => void;
   error: string;
 };
 
-function Register(props: RegisterProps) {
-  const { loading, onNext, onClickTerms, onClickPrivacyPolicy } = props;
+const MIN_LENGTH = 8;
 
-  const minLength = 8;
-  const [newPassword, setNewPassword] = useState("");
-  const [confirmPassword, setConfirmPassword] = useState("");
-  const [hint, setHint] = useState(`Minimum length: ${minLength}`);
+function passwordError(password: string): string | undefined {
+  if (password.length < MIN_LENGTH)
+    return `Must be at least ${MIN_LENGTH} characters`;
+  return;
+}
 
-  const onClick = () => {
-    if (arePasswordsValidAndEquals && !loading) {
-      onNext(newPassword);
-    }
-  };
+function confirmError(password: string, confirm: string): string | undefined {
+  if (confirm !== password) return "Confirmation does not match";
+  return;
+}
 
-  const isNewPasswordEmpty = newPassword.length === 0;
-  const isConfirmPasswordEmpty = confirmPassword.length === 0;
-  const isNewPasswordValid = newPassword.length >= minLength;
-  const isConfirmPasswordValid = confirmPassword.length >= minLength;
-  const arePasswordsValid =
-    !isNewPasswordEmpty &&
-    isNewPasswordValid &&
-    !isConfirmPasswordEmpty &&
-    isConfirmPasswordValid;
-  const arePasswordsEquals = newPassword === confirmPassword;
-  const arePasswordsValidAndEquals = arePasswordsValid && arePasswordsEquals;
-
-  const onChangeNewPassword = ({
-    target: { value },
-  }: React.ChangeEvent<HTMLInputElement>) => {
-    setNewPassword(value);
-
-    if (value.length > 0) {
-      setHint("");
-      return;
-    }
-
-    setHint(`Minimum length: ${minLength}`);
-  };
-  const onChangeConfirmPassword = ({
-    target: { value },
-  }: React.ChangeEvent<HTMLInputElement>) => {
-    setConfirmPassword(value);
+function Register({ loading, onNext }: RegisterProps) {
+  const [newPassword, setNew] = useState("");
+  const [confirm, setConfirm] = useState("");
+
+  const newErr = passwordError(newPassword);
+  const confirmErr = confirmError(newPassword, confirm);
+
+  const valid = newErr == null && confirmErr == null;
```

Comment:
`passwordError` returns `undefined` if the password is valid. This comparison works because `undefined == null` returns `true`. What do you think of having `passwordError` return `null` and using `===` here and elsewhere?

Reply:
It's very common to use `== null` to check for null and undefined. If we use undefined, we can pass the values directly to `error` on the child component (null isn't a valid argument to property of type `string|undefined`). The only time `==` is used in modern TypeScript / JavaScript is when checking for undefined/null. Otherwise, the coercions are too surprising.